self-core-impl/src/test/java/com/selfxdsd/core/managers/StoredProjectManagerTestCase.java

```diff
@@ -125,6 +126,140 @@ public void returnsProjects() {
+
+        final Storage storage = Mockito.mock(Storage.class);
+        Mockito.when(storage.projects()).thenReturn(all);
+
+        ProjectManager ProjectManager = new StoredProjectManager(1,
+                "123",
+                "zoeself",
+                Provider.Names.GITLAB,
+                "123token",
+                storage);
+        User PmUser = new StoredProjectManager(ProjectManager.id(),
+                ProjectManager.userId(),
+                ProjectManager.username(),
+                Provider.Names.GITLAB,
+                "123token",
+                storage).new PmUser(ProjectManager);
+
+        MatcherAssert.assertThat(
+                PmUser.role(),
+                Matchers.is("user")
+        );
+    }
+
+    /**
+     * PmUser returns its assigned projects.
+     */
+    @Test
+    public void pmUserReturnsProjects() {
+        final Projects assigned = Mockito.mock(Projects.class);
+        final Projects all = Mockito.mock(Projects.class);
+        Mockito.when(all.assignedTo(1)).thenReturn(assigned);
+
+        final Storage storage = Mockito.mock(Storage.class);
+        Mockito.when(storage.projects()).thenReturn(all);
+
+
+        ProjectManager ProjectManager = new StoredProjectManager(1,
+                "123",
+                "zoeself",
+                Provider.Names.GITLAB,
+                "123token",
+                storage);
+
+        User PmUser = new StoredProjectManager(ProjectManager.id(),
+                ProjectManager.userId(),
+                ProjectManager.username(),
+                Provider.Names.GITLAB,
+                "123token",
+                storage).new PmUser(ProjectManager);
+
+        MatcherAssert.assertThat(
+                PmUser.projects(),
+                Matchers.is(assigned)
+        );
+
+    }
+
+    /**
+     * PmUser returns its provider.
+     */
+    @Test
+    public void pmUserReturnsProvider() {
+        final Projects assigned = Mockito.mock(Projects.class);
+        final Projects all = Mockito.mock(Projects.class);
+        Mockito.when(all.assignedTo(1)).thenReturn(assigned);
+
+        final Storage storage = Mockito.mock(Storage.class);
+        Mockito.when(storage.projects()).thenReturn(all);
+
+
+        ProjectManager ProjectManager = new StoredProjectManager(1,
+                "123",
+                "zoeself",
+                Provider.Names.GITLAB,
+                "123token",
+                storage);
+
+        User PmUser = new StoredProjectManager(ProjectManager.id(),
+                ProjectManager.userId(),
+                ProjectManager.username(),
+                Provider.Names.GITLAB,
+                "123token",
+                storage).new PmUser(ProjectManager);
+
+        MatcherAssert.assertThat(
+                PmUser.provider(),
+                Matchers.allOf(
+                        Matchers.notNullValue(),
+                        Matchers.instanceOf(Gitlab.class)
+                )
+        );
+    }
```

Comment:
@HoreaPetrila 

```java
    final Provider provider = Mockito.mock(Provider.class);
    final ProjectManager manager = Mockito.mock(ProjectManager.class);
    Mockito.when(manager.provider()).thenReturn(provider);
    final User pmUser = new StoredProjectManager.PmUser(manager);

    MatcherAssert.assertThat(
        pmUser.provider(),
        Matchers.is(provider)
    );
```

However, there is a bug in PmUser.provider(): 
https://github.com/self-xdsd/self-core/blob/026d9626fec937adbcb9bf96c216f780bb140434/self-core-impl/src/main/java/com/selfxdsd/core/managers/StoredProjectManager.java#L252

It returns ``StoredProjectManager.this.provider()``. It should return ``this.manager.provider()`` -- can you fix it please?